src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -304,9 +304,93 @@ public void TestStartOnOSXWithBadFormat()
             }
         }
 
+        [Fact]
+        public void TestStartWithNonExistingUserThrows()
+        {
+            Process p = CreateProcessPortable(RemotelyInvokable.Dummy);
+            p.StartInfo.UserName = "DoesNotExist";
+            Assert.Throws<Win32Exception>(() => p.Start());
+        }
+
+        /// <summary>
+        /// Tests when running as a normal user and starting a new process as the same user
+        /// works as expected.
+        /// </summary>
+        [Fact]
+        public void TestStartWithNormalUser()
+        {
+            TestStartWithUserName();
+        }
+
+        /// <summary>
+        /// Tests when running as root and starting a new process as a normal user,
+        /// the new process doesn't have elevated privileges.
+        /// </summary>
+        [Fact]
+        [OuterLoop("Needs sudo access")]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
```

Comment:
Seems ideally tests.targets would pass `-notrait requireselevation` iff it's not running with sudo access. OuterLoop and sudo seem orthogonal

Reply:
The reason for `OuterLoop` is so it doesn't run normally on dev machines.  We do this similar pattern elsewhere:  

https://github.com/dotnet/corefx/blob/4bf90198c5e7710ea67b7c1b467e859ec8bb78e5/src/System.IO.FileSystem/tests/File/Delete.cs#L127-L129
https://github.com/dotnet/corefx/blob/4bf90198c5e7710ea67b7c1b467e859ec8bb78e5/src/System.IO.FileSystem/tests/Directory/Delete.cs#L206-L208
https://github.com/dotnet/corefx/blob/bf412174f02f751445df31a6393417d4eece5a16/src/System.Net.Security/tests/FunctionalTests/NegotiateStreamTest.Linux.cs#L118  (all the tests are marked with `OuterLoop`)